File: Bots/DotNet/Skills/CodeFirst/WaterfallSkillBot/Dialogs/Cards/CardOptions.cs
﻿namespace Microsoft.BotFrameworkFunctionalTests.WaterfallSkillBot.Dialogs.Cards
{
    public enum CardOptions
    {
        /// <summary>
        /// Adaptive card - Bot action
        /// </summary>
        BotAction,
        
        /// <summary>
        /// Adaptive card - Task module
        /// </summary>
        TaskModule,

        /// <summary>
        /// Adaptive card - Submit action
        /// </summary>
        SumbitAction,

        /// <summary>
        /// Hero cards
        /// </summary>
        Hero,
        
        /// <summary>
        /// Thumbnail cards
        /// </summary>
        Thumbnail,

        /// <summary>
        /// Receipt cards
        /// </summary>
        Receipt,

        /// <summary>
        /// Signin cards
        /// </summary>
        Signin,

        /// <summary>
        /// Carousel cards
        /// </summary>
        Carousel,

        /// <summary>
        /// List cards
        /// </summary>
        List,

        /// <summary>
        /// O365 cards
        /// </summary>
        O365,

        /// <summary>
        /// File cards
        /// </summary>
        File,

        /// <summary>
        /// Animation cards
        /// </summary>
        Animation,

        /// <summary>
        /// Audio cards
        /// </summary>
        Audio,

        /// <summary>
        /// Video cards
        /// </summary>
        Video,

        /// <summary>
        /// UploadFile cards
        /// </summary>
        UploadFile,

        /// <summary>
        /// Ends the card selection dialog
        /// </summary>
        End
    }
}

File: Bots/Python/Skills/CodeFirst/WaterfallSkillBot/dialogs/cards/card_options.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from enum import Enum


class CardOptions(str, Enum):
    adaptive_card_bot_action = "AdaptiveCardBotAction"
    adaptive_card_task_module = "AdaptiveCardTaskModule"
    adaptive_card_sumbit_action = "AdaptiveCardSumbitAction"
    hero = "Hero"
    thumbnail = "Thumbnail"
    receipt = "Receipt"
    signin = "Signin"
    carousel = "Carousel"
    list = "List"
    o365 = "O365"
    teams_file_consent = "TeamsFileConsent"
    animation = "Animation"
    audio = "Audio"
    video = "Video"
    end = "End"
